boa_engine/src/builtins/intl/date_time_format.rs

```diff
@@ -114,3 +114,112 @@ impl DateTimeFormat {
         Ok(date_time_format.into())
     }
 }
+
+/// The abstract operation `toDateTimeOptions` is called with arguments `options`, `required` and
+/// `defaults`.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma402/#sec-todatetimeoptions
+#[cfg(test)]
+pub(crate) fn to_date_time_options(
+    options: &JsValue,
+    required: &str,
+    defaults: &str,
+    context: &mut Context,
+) -> JsResult<JsObject> {
```

Comment:
From what I can see about the callers of this abstract operation, `required` and `defaults` can only be either `date`, `time` or `any/all`. This would be better expressed as an enum. You can express `any/all` as a single `AnyAll` variant.